schemas/telemetry/sync/sync.4.schema.json

```diff
@@ -1,6 +1,512 @@
 {
   "$schema": "http://json-schema.org/draft-04/schema#",
-  "properties": {},
-  "title": "placeholder_schema",
+  "additionalProperties": false,
+  "description": "Schema for sync pings as sent by this code: https://searchfox.org/mozilla-central/rev/ee40541496d3ad738097eebadaf4965ca1343b7a/services/sync/modules/telemetry.js#490 and documented at https://firefox-source-docs.mozilla.org/toolkit/components/telemetry/telemetry/data/sync-ping.html",
+  "properties": {
+    "application": {
+      "$comment": "The application section is included here inline due to incompatible incoming data",
+      "additionalProperties": false,
+      "properties": {
+        "architecture": {
+          "type": "string"
+        },
+        "buildId": {
+          "$comment": "Note that we drop the pattern requirement commonly used by other ping types for this field due to observed values in incoming data.",
+          "type": "string"
+        },
+        "channel": {
+          "type": "string"
+        },
+        "displayVersion": {
+          "type": "string"
+        },
+        "name": {
+          "type": "string"
+        },
+        "platformVersion": {
+          "type": "string"
+        },
+        "vendor": {
+          "type": "string"
+        },
+        "version": {
+          "type": "string"
+        },
+        "xpcomAbi": {
+          "type": "string"
+        }
+      },
+      "required": [
+        "architecture",
+        "channel",
+        "name",
+        "platformVersion",
+        "version"
+      ],
+      "type": "object"
+    },
+    "creationDate": {
+      "$comment": "Note that we drop the pattern requirement commonly used by other ping types for this field due to observed values in incoming data.",
+      "type": "string"
+    },
+    "id": {
+      "pattern": "^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$",
+      "type": "string"
+    },
+    "payload": {
+      "$schema": "http://json-schema.org/draft-04/schema#",
+      "description": "schema for Sync pings, documentation avaliable in toolkit/components/telemetry/docs/sync-ping.rst",
+      "properties": {
+        "deviceID": {
+          "pattern": "^[0-9a-f]{64}$",
+          "type": [
+            "string",
+            "null"
+          ]
+        },
+        "discarded": {
+          "minimum": 1,
+          "type": "integer"
+        },
+        "events": {
```

Comment:
This section is dropped because of variant types in the nested array.

Reply:
Hopefully we can later handle events the same way we do in main & event pings (per https://github.com/mozilla/jsonschema-transpiler/issues/38)